provider/in_place_upgrade_base.py

```diff
@@ -0,0 +1,169 @@
+import re
+import time
+import logging
+
+
+from virttest import utils_package
+
+LOG_JOB = logging.getLogger('avocado.test')
+
+
+class IpuTest(object):
+    """
+    Class for in_place_upgrade test in the vm
+    """
+
+    def __init__(self, test, params, env):
+        """
+        Init the default values of in_place_upgrade object.
+
+        :param params: A dict containing VM preprocessing parameters.
+        :param env: The environment (a dict-like object).
+        """
+        self.session = None
+        self.test = test
+        self.params = params
+        self.env = env
```

Comment:
Seems it isn't used.

Reply:
We do need them, if we don't have we will have the following issues
 FAIL: Failed to create ipuser : 'IpuTest' object has no attribute 'params'

Reply:
So could you explain why env will effect the params?

Reply:
So let use remove env only, is that ok ? And let us keep the rest ?